pkg/cmd/server/etcd/etcd.go

```diff
@@ -135,7 +135,7 @@ func TestNewEtcdClient(etcdClient newetcdclient.Client) error {
 // NeverRetryOnFailure is a retry function for the etcdClient.  If there's only one machine, master election doesn't make much sense,
 // so we don't bother to retry, we simply dump the failure and return the error directly.
 func NeverRetryOnFailure(cluster *etcdclient.Cluster, numReqs int, lastResp http.Response, err error) error {
-	if len(cluster.Machines) > 1 {
+	if len(cluster.Machines) > 1 || numReqs < 2 {
```

Comment:
> This isn't for merge, this is smoking out whether the code that removed the retry loop that the server uses is fundamentally flawed.

At a great expense of time, I tracked it down to here last time:  https://github.com/openshift/origin/blob/master/Godeps/_workspace/src/github.com/coreos/etcd/wal/wal.go#L398-L408.  Since there's prometheus metrics on this, I'd suggest gathering them and seeing if this is still the spot with spikes.